modules/signals/spec/signal-store.spec.ts

```diff
@@ -268,32 +268,58 @@ describe('signalStore', () => {
       expect(message).toBe('onDestroy');
     });
 
-    // FIX: injection context will be provided for `onDestroy` in a separate PR
-    // see https://github.com/ngrx/platform/pull/4196#issuecomment-1875228588
     it('executes hooks in injection context', () => {
```

Comment:
```suggestion
    it('executes hooks factory in injection context', () => {
```

Additionally, we can add a test to ensure that `onInit` is also executed within the injection context for the first (old) signature.